typesafe/src/main/scala/zio/config/typesafe/TypeSafeConfigSource.scala

```diff
@@ -114,15 +113,37 @@ object TypeSafeConfigSource {
                 valueType <- effect(parentConfig.getValue(head).valueType())
                 res <- if (valueType == ConfigValueType.LIST) {
                         for {
-                          listOfConfig <- effect(parentConfig.getConfigList(head).asScala.toList)
-                          res <- seqEither(listOfConfig.map(eachConfig => loop(eachConfig, next, nextPath :+ head)))
-                                  .map(t => t.flatMap(_.toList))
-                                  .flatMap({
-                                    case Nil =>
-                                      Left(singleton(ReadError.missingValue[Vector[String], String](path)))
-                                    case h :: t => Right(::(h, t))
-                                  })
+                          // A few extra error handling.
+                          res <- effect(parentConfig.getConfigList(head).asScala.toList) match {
+                                  case Right(allConfigs) =>
+                                    seqEither(allConfigs.map(eachConfig => loop(eachConfig, next, nextPath :+ head)))
+                                      .map(t => t.flatMap(_.toList))
+                                      .flatMap({
+                                        case Nil =>
+                                          Left(singleton(ReadError.missingValue[Vector[String], String](path)))
+                                        case h :: t => Right(::(h, t))
+                                      })
 
+                                  case Left(_) =>
+                                    effect(parentConfig.getList(head).asScala.toList).flatMap {
+                                      {
+                                        case Nil =>
+                                          Left(singleton(ReadError.missingValue[Vector[String], String](path)))
+
+                                        case h :: t
+                                            if (::(h, t).forall(
+                                              t =>
+                                                t.valueType() != ConfigValueType.NULL ||
+                                                  t.valueType() != ConfigValueType.LIST ||
+                                                  t.valueType() != ConfigValueType.OBJECT
+                                            )) =>
+                                          Right(::(h.unwrapped().toString, t.map(_.unwrapped().toString)))
+
+                                        case _ => Left(singleton(ReadError.missingValue[Vector[String], String](path)))
+
+                                      }
+                                    }
+                                }
                         } yield res
                       } else {
                         if (parentConfig.getValue(head).valueType() == ConfigValueType.OBJECT) {
```

Comment:
I will need to revisit, however, there is just about lots of `Object` operations in type safe library. So any checks here probably make it much more robust.